aria/orchestrator/workflows/api/task.py

```diff
@@ -137,6 +137,11 @@ def __init__(self,
         operation = self.actor.interfaces[self.interface_name].operations[self.operation_name]
         self.plugin = operation.plugin
         self.function = operation.function
+
+        modeling_utils.validate_required_inputs_are_supplied(
+            declared_inputs=operation.inputs,
+            supplied_inputs=operation.arguments)
```

Comment:
After consulting with @tliron and reviewing the TOSCA spec, I came to the conclusion that the validation of the `required` field of inputs that belong to operations and interfaces (as opposed to topology template and workflow inputs) should be done in the parsing stage.
This reasoning follows the TOSCA spirit, where anything that is declared in the type and is required, must be assigned in the corresponding template.